### DISCUSSION TITLE
Forum Locked out due to EX1146

### DISCUSSION THREAD

**Discussion by ColdJ-KSP** [net-lisias-ksp/KSP-Forum-Preservation-Project#13] [General]
First I will quote you @Lisias , due to you already posting on reddit the gist of what I was going to say. Or was it Scarecrow 71?

[Scarecrow_71](https://www.reddit.com/user/Scarecrow_71/)
KSP Forums...And The Lack Thereof

Before we get started, I'm going to point out that I am NOT asking what the status of the forums is. They are down, and have been for days. [u/Vanamond3](https://www.reddit.com/user/Vanamond3/) has indicated that they have been in contact with the tech person, and that the tech person has informed them that the issue is being looked at. So this post has nothing to do with asking for status or to ask if anyone else has noticed that the forums are down.

Now, with that out of the way...

If you try to click on anything on the forums, you get a nice little page that says "Sorry, but something has gone wrong". And in the lower right-hand corner of the page (or, rather, about 1/3 of the way down the page, on the right-hand side) you will see an Error Code. EX1146, to be precise. I have to believe that the tech person is at least aware of this Error Code; all they would have to do is navigate to the forums, click on any link, and they'd see it. Right?

A simple Google search "Invision Error Code EX1146" returns a whole host of answers. And most of them indicate that one or more administrative tables critical to the health of the site is either corrupted or deleted, and that the solution is to recreate the table and its data from a recent backup of the site. In fact, one of the Invision employees on their own forum several years ago indicates this is the reason for the error:

[EX1146 - Can't get access to admin panel - Technical Problems - Invision Community](https://invisioncommunity.com/forums/topic/463344-ex1146-cant-get-access-to-admin-panel/#comment-2866717)

NOW
My brain is the analyst, big picture sort of model, rather than the technical/programmer sort ,so on reading the limited info available, my best guess is that when they renewed the the forum license for 6 months, that they did so with a smaller DATA LIMIT, which was immediately exceeded. (As they no longer had a vested interest they just did the bare minimum to placehold the forum.)
As a result you get the same glitch that happened to that Forum runner some years ago.
So unless the person with Admin access actually cares and puts in the effort, we are back to having to create an alternative forum and need a way to be access the archived forum data, like being able to read historical texts and reference books. Being able to copy and paste the relevant data from the help threads and mod threads, while leaving out all the comment posts that just bloat those threads.

So, any thoughts on what might be done?

**Comment by Lisias** [comment]
I don't think this last outage is related to Invision. When the license expires, the Invision Forum continues to work - it only loses access to Updates and Support.

Source: https://www.theadminzone.com/threads/ipb-now-invision-community-change-of-conditions.155351/post-1186015

So, nope. Whatever is happening, **it's not** related to the license - **UNLESS** they are using something else that also needs a license and it's not from Invision.

As I explained in your github repo here https://github.com/ColdJ-KSP/KSP-Locked/issues/1#issuecomment-2423913931 , the `1146` is a MySQL Exception and it means "could not read that damned table/view" but without really telling the reason:

* It may be because the table/view doesn't exists
* But it can also be because you weren't GRANTed permission to access it
* Or, sometimes, because the table/view got corrupted and is now unreadable.
* And perhaps some other possible problems

The most common reason for getting the `EX1146` is when a table called `ibf_core_log` gets unreadable by any of the reasons above - but, frankly, the fix is incredibly straightforward: just recreate it, removing it if it still exists first.

So, if Forum is still down, it should be due one of the following possible reasons (that I can think of at this moment, at least):

* There's no one left on whatever is caring for the Forum infrastructure on TTWO that knows how to handle Invision
* There's no one there **willing** to touch this thing
* There're something else **way bigger** happening, and KSP Forum was only one of the victims, and these poor bastards are buried in deep shit trying to fix the mess, and KSP Forum will come back after they clean things up.

I still think that Forum's "soul" is fine and sound, because that Digest mail I received 13 hours after Forum gone titties up - had any problem be happening on the profiles, topics or threads tables, that process would had fail. Same for the DBMS itself (MySQL) - had MySQL be inoperative, that process would not had sent me that Digest email.

Right now, this is all I can say about without risking trash talking (like some people are doing with me, by the way - just heard people blaming me for the outage due this work...). So, unfortunately, we will need to wait and see what happens.

**Comment by Lisias** [comment]
In time, since we restablish contact here, I'm deleting some comments of yours on some unrelated places. Please don't take it bad, it's just house keeping. :)

**Comment by ColdJ-KSP** [comment]
Did you notice that you can still get to the Guidelines page, the Staff page and the Users online page, even though the last one has spurious data?

**Reply by Lisias** [reply]
Yes. This is probably due caching. Something like this: https://memcached.org/ or perhaps https://dev.mysql.com/doc/refman/5.7/en/query-cache.html

The Front End itself also appears to be working (obviously). So the hosting machine wasn't hacked neither defaced, like happened with Internet Archive.

Right now, my best guess is that something bigger happened, and no one fixed Forum because they are buried in deep shit at this moment, and can't spare the time with something low priority.

**Reply by ColdJ-KSP** [reply]
Ok.
Well, I am only good for guessing. Got to go get other things done. Will check for new posts in my night time.
Good luck and hope your day goes well.

**Comment by Lisias** [comment]
You know something? I'm missing Forum. 😕

**Comment by towermom9** [comment]
imagine if it's just an accidental rm -rf 💀

**Reply by Lisias** [reply]
No risk. I received a Digest email about 13 to 14 hours after the sinister event. This strongly suggests that both the MySQL Server and the Forum's backend were working fine at that moment, ruling out any really nasty damage.

At least, until that moment.

**Reply by towermom9** [reply]
> No risk. I received a Digest email about 13 to 14 hours after the sinister event. This strongly suggests that both the MySQL Server and the Forum's backend were working fine at that moment, ruling out any really nasty damage.
> 
> At least, until that moment.

maybe the email service is on another server?

**Reply by Lisias** [reply]
Probably. But, still, it had to access the MySQL server to check for the threads and profiles I'm following, not to mention accessing **my** profile to fetch my email address.

And the Front Page **is** working. So, even if someone did a `rm  -rf` on some machine, it didn't destroyed the hosting the MySQL tables, where the Forum's "soul" is.

You can destroy anything else, you can get Forum back with some work. The only thing that would kill Forum for sure is destroying the MySQL tables (where the topics, threads, posts and profiles are stored).

**Comment by Nyxilo** [comment]
I don't know much about how a website works, but do I assume it's bad news that we're now receiving a 500 HTML error instead of a 502 or the more recent EX1146 ?

It looks like it's being triggered by the server hosting the forums, not by the forums themselves (unlike the previous errors we were getting in previous months).

Am I right? I hope not.

**Reply by Lisias** [reply]
Are you talking about this, right?

![Screen Shot 2024-10-22 at 15 14 32](https://github.com/user-attachments/assets/bdf95e36-b839-43b5-b7f5-c9411e0c58b4)

```
HTTP/1.1 500 Internal Server Error
date: Tue, 22 Oct 2024 18:15:41 GMT
content-type: text/html; charset=UTF-8
cf-ray: 8d6b795ffd1b4ecb-GRU
cf-cache-status: DYNAMIC
cache-control: no-cache, no-store, must-revalidate, max-age=0, s-maxage=0
expires: 0
cf-apo-via: origin,host
x-powered-by: PHP/8.1.19
vary: Accept-Encoding
server: cloudflare
X-Firefox-Spdy: h2

<!DOCTYPE html>
<html lang="en">
<head>
<title>Error</title>
<style type="text/css">
			body {
				background: #f9f9f9;
				margin: 0;
				padding: 30px 20px;
				font-family: "Helvetica Neue", helvetica, arial, sans-serif;
			}

			#error {
				max-width: 800px;
				background: #fff;
				margin: 0 auto;
			}

			h1 {
				background: #151515;
				color: #fff;
				font-size: 22px;
				font-weight: 500;
				padding: 10px;
			}

				h1 span {
					color: #7a7a7a;
					font-size: 14px;
					font-weight: normal;
				}

			#content {
				padding: 20px;
				line-height: 1.6;
			}

			#reload_button {
				background: #151515;
				color: #fff;
				border: 0;
				line-height: 34px;
				padding: 0 15px;
				font-family: "Helvetica Neue", helvetica, arial, sans-serif;
				font-size: 14px;
				border-radius: 3px;
			}
		</style>
</head>
<body>
<div id="error">
<h1>An error occurred <span>(500 Error)</span></h1>
<div id="content">
We're sorry, but a temporary technical error has occurred which means we cannot display this site right now.
<br><br>
You can try again by clicking the button below, or try again later.
<br><br>
<button onclick="window.location.reload();" id="reload_button">Try again</button>
</div>
</div>
</body>
</html>
```

**Reply by Lisias** [reply]
This means that someone pulled the plug on the Forun's Front Page.

It may mean anything - from the final demise to the first evidences they are working to fix Forum.

We need to be patient, wait and see.

**Reply by Nyxilo** [reply]
Okay, thanks for your response

**Comment by Lisias** [comment]
Dakota was reached by some people, and reply back on Twitter/X:

>kerbal nerds: had a bunch of people poke me about the forums being down
> 
> it seems like a small configuration issue, so with the proper changes, there shouldn't be any data loss
>
> i've passed along your messages to the people at T2 who can fix it, but the ball is in their court now

https://x.com/DakotaCxc/status/1848907185465819210

And Vanamonde also hit Reddit again about this subject:

> The contact we contacted has contacted his contact who has now been granted access to do something to something. 

https://www.reddit.com/r/KerbalSpaceProgram/comments/1gannrg/comment/ltfnqko/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

**Comment by Lisias** [comment]
Forum is (finally) back.

But I will keep this opened for some days - just in case.

**Comment by Lisias** [comment]
At this time, with PD with a new owner and some investments made on the Forum's infrastructure, it's unlikely that Forum will be shutdown in the near (or even long) future.

So I'm closing this.

**Reply by ColdJ-KSP** [reply]
Woops :)

**Reply by Lisias** [reply]
It's not closing, it's the other way around. They are moving servers around the world.

In a few hours, the new DNS entry will propagate around the World.

**Reply by Lisias** [reply]
One DNS entry already propagated into my home network, but the second entry wasn't created yet.

I'm starting to wonder why...
